src/integer/z/from.rs

```diff
@@ -67,6 +68,29 @@ impl Z {
     from_type!(u32, u64, Z, Z::from_u64);
     from_type!(u16, u64, Z, Z::from_u64);
     from_type!(u8, u64, Z, Z::from_u64);
+
+    /// Create a new Integer that can grow arbitrary large.
+    ///
+    /// Parameters:
+    /// - `value`: the initial value the integer should have
+    ///
+    /// Returns the new integer.
+    ///
+    /// # Example
+    /// ```
+    /// use math::integer::Z;
+    /// use math::integer_mod_q::Modulus;
+    /// use std::str::FromStr;
+    ///
+    /// let m = Modulus::from_str("42").unwrap();
+    ///
+    /// let a: Z = Z::from_modulus(m);
+    /// ```
+    pub fn from_modulus(value: Modulus) -> Self {
+        Z {
+            value: value.get_fmpz_mod_ctx_struct().n[0],
```

Comment:
memory leak: directly added the fmpz. If either the Modulus is dropped. The fmpz value would also be dropped. Hence if the value is large (a pointer, we cause a memory leak). 
```suggestion
            let mut out = Z::default();
            unsafe{fmpz_set(&mut out.value, &value.get_fmpz_mod_ctx_struct().n[0]};
            out
```